add_slot_command

chejinge opened the pull request.

luky116 committed:
Merge pull request #6 from luky116/newCode2

fix: remove new/delete in net (#1503)

luky116 committed:
feat: merge integ test (#8)

* add integrate test

luky116 committed:
feat: merge integ test (#8)

* add integrate test

chejinge committed:
delete_sharding_mode

chejinge committed:
delete_sharding_mode (#10)

chejinge committed:
delete_sharding_mode (#10)

chejinge committed:
add_slot_command

chejinge committed:
Update pika_conf.cc

chejinge committed:
Merge branch 'unstable' into unstable

chejinge committed:
Update pika.yml

chejinge committed:
Merge pull request #13 from chejinge/pikiwi/rebase

rebase to upstream

chejinge committed:
Merge branch 'pikiwidb:unstable' into unstable

chejinge committed:
Update pika_slot.cc

chejinge committed:
bugfix

chejinge committed:
point->smartpoint

chejinge committed:
Merge pull request #14 from chejinge/unstable

add_slot_command_to_codis

Commit:
resolve conflict

AlexStocks committed:
Merge pull request #20 from luky116/mergeCode2

merge unstable code

chejinge committed:
std::shared_ptr<Partition>partition->std::shared_ptr<Slot>slot

chejinge committed:
referator class slot nocopable

AlexStocks committed:
Merge pull request #25 from chejinge/unstable

std::shared_ptr<Partition>partition->std::shared_ptr<Slot>slot

chejinge committed:
partition->slot

AlexStocks committed:
Merge pull request #26 from chejinge/unstable

partition->slot

chejinge committed:
partition->slot

chejinge committed:
tools reverst

chejinge committed:
Db->DB

AlexStocks committed:
Merge pull request #27 from chejinge/unstable

partition->slot

chejinge committed:
table->db

chejinge committed:
table->db

chejinge committed:
cmddb->cmdtable

chejinge committed:
table->db

AlexStocks committed:
Merge pull request #28 from chejinge/unstable

table->db

chejinge committed:
add_codis_slot_command

chejinge committed:
add_slot_command

chejinge committed:
reslove_build_bug

chejinge committed:
add table_db

chejinge committed:
fix

chejinge committed:
fix

chejinge committed:
add_slot_debug

chejinge committed:
Merge pull request #30 from chejinge/unstable

[WIP]debug_slot_command

Commit:
add key to slot sets

Commit:
add key to slot sets

Commit:
fix

Commit:
fix

chejinge committed:
Merge pull request #31 from luky116/debugSlotCmmand

writer command to add key to slot sets

Commit:
fix slotsmgrt_sender_thread_ bug

Commit:
Merge remote-tracking branch 'piki/debugSlotCmmand' into tmplocal

Commit:
add pika megrate thread

chejinge committed:
fix SLOTSMGRTTAGSLOT (#33)

Commit:
支持多线程进行迁移slot

luky116 committed:
fix slotsmgrt_sender_thread_ bug (#32)

fix slotsmgrt_sender_thread_ bug

luky116 committed:
support multi-threaded migration slots (#37)

support multi-threaded migration slots

chejinge committed:
feat: add SlotsDel and SlotsHashKey cmd (#38)

feat: add SlotsDel and SlotsHashKey cmd

Commit:
Fix lock conflicts in multithreading

Commit:
fix conflict

Commit:
remove comment

Commit:
add todo comment

chejinge committed:
reslove conflict

chejinge committed:
reslove_conflict

chejinge committed:
reslove_build_bug

Commit:
rename func

chejinge committed:
fix_info_command

Commit:
set running_ atomic bool

Commit:
fast failed

Commit:
use func

AlexStocks committed:
Merge pull request #39 from luky116/fixLocks

Fix lock conflicts in multi threading

chejinge committed:
switch_code_style

Commit:
add slotsmgrt-exec-wrapper

chejinge committed:
fix_pika_conf

chejinge committed:
fix_code_note

chejinge committed:
Merge pull request #42 from chejinge/mergecode

MergeAtomcode->pikiwidb

Commit:
fast failed

chejinge committed:
Merge pull request #44 from luky116/addSlotsmgrt-exec-wrapper

add slotsmgrt-exec-wrapper command

chejinge committed:
fix_info_command&&addslots_scan_command (#46)

* fix_info_command&&addslots_scan_command

* change virtual to overwrite

* string2int->std::stoll

* fix_command

luky116 committed:
fix slots-hash-key cmd bug (#49)

fix SlotsHashKeyCmd bug

Co-authored-by: liuyuecai <liuyuecai@360.cn>

luky116 committed:
feat: add slot migrate cmd (#53)

add slot migrate cmd

Commit:
fix conflict

luky116 committed:
merge unstable code

merge unstable code

luky116 committed:
fix: rename function GetSlotID

rename function GetSlotID name

chejinge committed:
fix_slot_info_command (#67)

luky116 committed:
optimize: remove unsed param (#83)

remove unsed param

luky116 committed:
optimize code style (#89)

optimize code style

chejinge committed:
slotsscan add param check (#94)

slotsscan add slot param range check

chejinge committed:
support config rewite slotmigrate param (#97)

support config rewite slotmigrate param (#97)

luky116 committed:
refactor async migrate (#95)

refactor async migrate

luky116 committed:
fix slotsmgrt-async-status command (#99)

fix slotsmgrt-async-status command

luky116 committed:
fix slotsmgrt-exec-wrapper command (#100)

change the return value of slotsmgrt-exec-wrapper command:

1. return 0 means key not exist or key is not migrating
2. return 1 means key is migrating
3. return -1 means something wrong

chejinge committed:
Slotclean slot reload (#85)

* add_reload_slot_command and cleanup_command

luky116 committed:
feat: add binlog when migrate (#59)

add binlog when migrate

chejinge committed:
fix_slot_command_code_style (#102)

* fix_code_style

luky116 committed:
set const std::shared_ptr param (#103)

luky116 committed:
feat: append binlog when add slot key (#106)

* append binlog when add slot key

luky116 committed:
fix problem (#111)

Co-authored-by: liuyuecai <liuyuecai@360.cn>

luky116 committed:
feat: slotscleanup cmd add binlog (#115)

* slots cleanup add binlog

luky116 committed:
optimize async migrate (#110)

* append binlog when add slot key

chejinge committed:
optimize code style (#117)

Commit:
resolve conflict

AlexStocks commented on conf/pika.conf:
```diff
@@ -113,7 +106,7 @@ consensus-level : 0
 
 # daemonize  [yes | no].
-#daemonize : yes
+daemonize : no
```
这行为何要改呢？如果是测试需要，可以在 .github 下面单独放置一个测试配置文件

chejinge replied:
好的

AlexStocks commented on include/pika_migrate_thread.h:
```diff
@@ -0,0 +1,112 @@
+
+ private:
+  std::string dest_ip_;
+  int64_t dest_port_;
```
class member 都得有初始化值

AlexStocks commented on include/pika_migrate_thread.h:
```diff
@@ -0,0 +1,112 @@
+  virtual void *ThreadMain();
+
+ private:
+  std::string dest_ip_;
```
ditto，class member 初始化值

AlexStocks commented on include/pika_server.h:
```diff
@@ -315,6 +327,127 @@ class PikaServer : public pstd::noncopyable {
+ * BGSlotsReload used
+   */
+  struct BGSlotsReload {
+    bool reloading;
```
给下初始化值吧

AlexStocks commented on include/pika_server.h:
```diff
@@ -315,6 +327,127 @@ class PikaServer : public pstd::noncopyable {
+   * BGSlotsCleanup used
+   */
+  struct BGSlotsCleanup {
+    bool cleaningup;
```
ditto, struct init member value

AlexStocks commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,282 @@
+  void Merge() override {};
+  Cmd* Clone() override { return new SlotsMgrtTagSlotCmd(*this); }
+ private:
+  std::string dest_ip_;
```
class init value

AlexStocks commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,282 @@
+  void Merge(){};
+  Cmd* Clone() override { return new SlotsMgrtTagSlotAsyncCmd(*this); }
+ private:
+  std::string dest_ip_;
```
class init value

AlexStocks commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,282 @@
+  Cmd* Clone() override { return new SlotsMgrtTagOneCmd(*this); }
+ private:
+  std::string dest_ip_;
+  int64_t dest_port_;
```
class init value

AlexStocks commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,282 @@
+ private:
+  void DoInitial() override;
+
+  int64_t begin_;
```
class init value

luky116 committed:
Merge pull request #119 from luky116/mergeCode-0619

merge unstable code

chejinge committed:
class_member init value (#120)

* class_member init value

AlexStocks commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,283 @@
+  void CleanMigrateClient();
+
+  void Lock() {
+    LOG(INFO) << "migrate lock";
```
留 info 级别日志干嘛？

chejinge committed:
fix_code_style (#122)

Mixficsol commented on include/pika_conf.h:
```diff
@@ -106,6 +114,14 @@ class PikaConf : public pstd::BaseConf {
     std::shared_lock l(rwlock_);
     return timeout_;
   }
+  int binlog_writer_num(){
```
）和 {之间空格

Mixficsol commented on include/pika_conf.h:
```diff
@@ -106,6 +114,14 @@ class PikaConf : public pstd::BaseConf {
+    std::shared_lock l(rwlock_);
+    return binlog_writer_num_;
+  }
+  bool slotmigrate(){
```
）和 {之间空格

Mixficsol commented on include/pika_command.h:
```diff
@@ -203,6 +221,9 @@ enum CmdFlagsMask {
   kCmdFlagsMaskSuspend = 64,
   kCmdFlagsMaskPrior = 128,
   kCmdFlagsMaskAdminRequire = 256,
+  kCmdFlagsMaskPreDo  = 512,
```
等号左右一个空格

Mixficsol commented on include/pika_conf.h:
```diff
@@ -388,6 +404,10 @@ class PikaConf : public pstd::BaseConf {
   }
+  void SetSlotMigrate(const std::string &value) {
+    std::lock_guard l(rwlock_);
+    slotmigrate_ =  (value == "yes") ? true : false;
```
等号左右一个空格

Mixficsol commented on include/pika_define.h:
```diff
@@ -12,6 +12,13 @@
+/*
+ * TTL type
+ */
+#define PIKA_TTL_ZERO     0
```
宏和数字之间也是一个空格，参照下面的

Mixficsol commented on include/pika_server.h:
```diff
@@ -316,6 +328,127 @@ class PikaServer : public pstd::noncopyable {
+   * * Async migrate used
+   */
+  int SlotsMigrateOne(const std::string &key, std::shared_ptr<Slot>slot);
+  bool SlotsMigrateBatch(const std::string &ip, int64_t port, int64_t time_out, int64_t slots,int64_t keys_num, std::shared_ptr<Slot>slot);
```
, int64_t keys_num空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsDelCmd : public Cmd {
+ public:
+  SlotsDelCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsHashKeyCmd : public Cmd {
+ public:
+  SlotsHashKeyCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsScanCmd : public Cmd {
+ public:
+  SlotsScanCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+* */
+class SlotsMgrtExecWrapperCmd : public Cmd {
+ public:
+  SlotsMgrtExecWrapperCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsReloadCmd : public Cmd {
+ public:
+  SlotsReloadCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsReloadOffCmd : public Cmd {
+ public:
+  SlotsReloadOffCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsCleanupCmd : public Cmd {
+ public:
+  SlotsCleanupCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on include/pika_slot_command.h:
```diff
@@ -0,0 +1,280 @@
+
+class SlotsCleanupOffCmd : public Cmd {
+ public:
+  SlotsCleanupOffCmd(const std::string& name, int arity, uint16_t flag):Cmd(name, arity, flag) {}
```
空格

Mixficsol commented on src/pika_admin.cc:
```diff
@@ -1724,7 +1731,10 @@ void ConfigCmd::ConfigSet(std::string& ret) {
+  } else if (set_item == "slotmigrate") {
+    g_pika_conf->SetSlotMigrate(value);
+    ret = "+OK\r\n";
+  }else if (set_item == "userblacklist") {
```
空格

chejinge committed:
fix_code_style (#123)

chejinge committed:
Fix code style (#124)

* reslove_confilict

luky116 committed:
resolve conflict (#125)

* [Unit Test]fix: maxmemory.tcl & printver.tcl #1624 (#1625)

* fix unit test:type/set

* 修复printver.tcl

* 修复maxmemory.tcl不支持的命令，新增一个config get的单测

* 添加注释信息与原redis相关测试用例

* tests/test_helper.tcl 取消注释

* Fixed the RpopLpush's replication problem between master and slave (#1616)

* 1.RpopLpush:lockd both key in Storage layer
2.Made RpopLpush write two binlog(rpop+lpush) instead of an binlog of RpopLpush

* add test to github workflow

* revised pika.yml

* move start script

---------

Co-authored-by: cjh <1271435567@qq.com>

---------

Co-authored-by: chenbt <34958405+chenbt-hz@users.noreply.github.com>
Co-authored-by: Junhua Chen <41671101+cheniujh@users.noreply.github.com>
Co-authored-by: cjh <1271435567@qq.com>
Co-authored-by: liuyuecai <liuyuecai@360.cn>
